Mathlib/Data/Matrix/Block.lean

```diff
@@ -699,7 +699,7 @@ theorem blockDiagonal'_pow [∀ i, DecidableEq (m' i)] [Fintype o] [∀ i, Finty
   map_pow (blockDiagonal'RingHom m' α) M n
 
 @[simp]
-theorem blockDiagonal'_smul {R : Type*} [Semiring R] [AddCommMonoid α] [Module R α] (x : R)
+theorem blockDiagonal'_smul {R : Type*} [Monoid R] [AddMonoid α] [DistribMulAction R α] (x : R)
```

Comment:
```suggestion
theorem blockDiagonal'_smul {R : Type*} [Zero α] [SMulZeroClass R α]  (x : R)
```

Reply:
Thanks, I'll add that to the list of custom generalizations I try to perform.